src/docs/sphinx/dev_guide/gpu_porting.rst

```diff
@@ -0,0 +1,297 @@
+    uses the loop_policy from the templated execution space.
+  * ``reduce_policy`` - For  `RAJA reduction types`_
+  * ``atomic_policy`` - For `RAJA atomic operations`_
+  * ``sync_policy`` - For Axom's `synchronize`_ function, which is a wrapper
+    around ``RAJA::synchronize()``
+
+* Umpire allocator defaults
+
+  * ``memory_space`` - The memory space abstraction for use by
+    :ref:`Core Containers<core-containers>` like ``axom::Array``.
+  * ``allocatorID()`` - Gets the allocator ID for the Umpire resource to use in
+    this execution space.
+
+* General information on the execution space
+
+  * ``name()`` - Name of the execution space
+  * ``onDevice()`` - Is the execution space on device? (True/False)
+  * ``valid()`` - Is the execution space valid? (True)
+  * ``async()`` - Is the execution space asynchronous? (True/False)
+
+There is also a set of `nested execution policies`_
+(axom/mint/execution/internal/structured_exec.hpp) to be used with
+``RAJA::kernel``, and are used primarily for iterating over mint meshes.
+
+
+.. _gpu-tips-label:
+
+================================
+General, Rough Porting Tips
+================================
+
+* Start with figuring out what memory you need on device, and use Axom's Array
+  class and memory_managment functions to do the allocations.
+* Using a barebones Axom's for_all kernel with a device policy, attempt to
+  access/ manipulate the memory on device.
+
+  * This does not have to be logically correct, you are just making sure you
+    have done the allocation correctly (e.g. not segfaulting).
+
+* Add the functions you want to call on device to the ``axom::for_all`` kernel.
+
+  * Does not have to be logically correct.
+
+* When you attempt to recompile your code, unless you have already done the
+  ``__host__ __device__`` decoration correctly already, the compiler should
+  complain about what functions are not yet compiling on device, and that you
+  need to add ``__host__ __device__`` decorators::
+
+    error: calling a __host__ function("XXX") from a __host__ __device__ function("YYY") is not allowed
+
+  * Even after you add the decorators to the functions the compiler complains
+    about, it is likely recompiling will introduce complaints about more
+    functions (the functions being the non-decorated functions your
+    newly-decorated functions are calling). You will want to keep decorating
+    until all the complaints are gone.
+  * Most of the C++ standard library is not available on device. Your options
+    are Axom's equivalent functions/classes if it exists, or to add your
+    own / rewrite the code to not use standard library.
+
+* With no more decorating complaints from the compiler, write the logically
+  correct kernel.
+
+  * If at this point your kernel is not working/segfaulting, it's hopefully a
+    logical error, and you can debug the kernel without diving into debugging
+    tools.
+  * ``printf()`` for output
```

Comment:
> * ``printf()`` for output

Please expand this line into a full sentence.

Also, should we mention which, if any, `slic` macros work in kernel code?